be/src/util/buffered_stream.h

```diff
@@ -64,4 +72,41 @@ class BufferedInputStream {
     uint64_t _file_offset = 0;
 };
 
+class SharedBufferedInputStream : public IBufferedInputStream {
+public:
+    struct IORange {
+        int64_t offset;
+        int64_t size;
+        bool operator<(const IORange& x) const { return offset < x.offset; }
+    };
+    struct CoalesceOptions {
+        static constexpr int64_t MB = 1024 * 1024;
+        int64_t max_read_size = 16 * MB;
+        int64_t max_dist_size = 1 * MB;
+        int64_t max_buffer_size = 8 * MB;
+    };
+
+    SharedBufferedInputStream(RandomAccessFile* file);
+
+    ~SharedBufferedInputStream() override = default;
+
+    Status set_io_ranges(const std::vector<IORange>& ranges);
+    Status remove_io_range(const IORange& range);
+
+    Status get_bytes(const uint8_t** buffer, size_t offset, size_t* nbytes) override;
+    void release();
+
+private:
+    struct SharedBuffer {
+    public:
+        int64_t offset;
+        int64_t size;
+        int64_t ref_count;
```

Comment:
I haven't find out where do you use this variable

Reply:
No need right now.